logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/selfDeletingMessages/SelfDeletionTimer.kt

```diff
@@ -22,30 +22,28 @@ import com.wire.kalium.util.serialization.toJsonElement
 import kotlin.time.Duration
 import kotlin.time.Duration.Companion.ZERO
 
-sealed class SelfDeletionTimer {
+sealed interface SelfDeletionTimer {
+    val duration: Duration?
+
     /**
      * Represents a self deletion timer that is currently disabled
      */
-    object Disabled : SelfDeletionTimer()
+    object Disabled : SelfDeletionTimer {
+        override val duration: Duration? = null
+    }
 
     /**
      * Represents a self deletion timer that is enabled and can be changed/updated by the user
      */
-    data class Enabled(val userDuration: Duration) : SelfDeletionTimer()
+    data class Enabled(override val duration: Duration?) : SelfDeletionTimer
```

Comment:
```suggestion
    data class Enabled(override val duration: Duration) : SelfDeletionTimer
```

Should it be nullable?

Reply:
Marking as resolved as I see that `Enabled` should have nullable durability as well.